frontend/javascripts/oxalis/view/context_menu.tsx

```diff
@@ -1090,15 +1188,29 @@ function ContextMenuInner(propsWithInputRef: Props) {
       );
     }
 
-    if (segmentIdAtPosition > 0) {
+    if (segmentIdAtPosition > 0 || maybeClickedMeshId != null) {
```

Comment:
Could it be that both are defined? Or is one case only for the data viewports and the second case only for the 3D viewport? If both were possible, the user wouldn't know which ID is shown, right?

Reply:
It is not possible that both are defined. The segmentId is calculated based upon the calculated global position of where the context menu is opened. The function used for that is [_calculateMaybeGlobalPos](https://github.com/scalableminds/webknossos/blob/b3673cdfa979ac882baf5a0b5b35c65f4b8f79f3/frontend/javascripts/oxalis/model/accessors/view_mode_accessor.ts#L98). Thus in case of the 3d viewport, the segmentId will never be defined.
And the `maybeClickedMeshI` can only be defined when the context menu is opened in the 3d viewport. Thus if on variable is defined / `> 0` the other cannot. They exclude each other 🎉 .


I am personally not perfectly happy with my solution here, but other alternatives would me more code duplication. Thus I added a comment shortly explaining that only one of them can be defined.

Reply:
> It is not possible that both are defined. The segmentId is calculated based upon the calculated global position of where the context menu is opened. The function used for that is [_calculateMaybeGlobalPos](https://github.com/scalableminds/webknossos/blob/b3673cdfa979ac882baf5a0b5b35c65f4b8f79f3/frontend/javascripts/oxalis/model/accessors/view_mode_accessor.ts#L98). Thus in case of the 3d viewport, the segmentId will never be defined.
And the maybeClickedMeshI can only be defined when the context menu is opened in the 3d viewport. Thus if on variable is defined / > 0 the other cannot. They exclude each other tada .

Great :+1: Thanks for the explanation!

> I am personally not perfectly happy with my solution here, but other alternatives would me more code duplication. Thus I added a comment shortly explaining that only one of them can be defined.

Yeah, I think, it's a good trade-off.